.github/workflows/build.yml

```diff
@@ -0,0 +1,54 @@
+name: build
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '*'
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      forcePublish:
+        description: When true, the PowerShell module will be published to specified PowerShell Repository irrespective of its version
+        required: true
+        default: false
+        type: boolean
+
+permissions:
+  checks: write  # enable test result annotations
+  contents: write  # enable creating releases
+  issues: read
+  packages: write  # enable publishing packages
+  pull-requests: write  # enable test result annotations
+
+jobs:
```

Comment:
It all starts to get a bit meta - but would the artifact caching (for PS Modules) help speed up the build?